modules/apps/bean-portlet/bean-portlet-cdi-extension/src/main/java/com/liferay/bean/portlet/cdi/extension/internal/BeanPortletInvokerPortlet.java

```diff
@@ -91,12 +95,19 @@ public BeanPortletInvokerPortlet(
 
 	@Override
 	public void destroy() {
+		if (_destroyed.get()) {
```

Comment:
@dantewang Have you observed multiple threads attempting to call this method? I suppose that the Aries CDI thread would be responsible for invoking the `@PreDestroy` methods, and then a different thread is responsible for undeploying the portlet. Perhaps that is why you felt it necessary to use an `AtomicBoolean`?

Reply:
@ngriffin7a No, I didn't pay attention to this area... I made this pull mainly to show the idea so I didn't check carefully, but I think I saw somewhere in Weld using `AtomicBoolean`.

Is this approach an acceptable solution for the issue?

Reply:
@dantewang Yes, this is a good/acceptable solution, but as I wrote in the main description of this issue, we *might* need a Spring-based equivalent. Won't know for sure until the MVCBean PR is merged and we retry the v3demo portlet with Spring instead of CI.

Reply:
@dantewang I think that the usage of `AtomicBoolean` is probably not necessary. I don't think that this problem manifests itself in a multithreaded situation where there would be a race condition. I could be wrong but that's my best guess. You would have to investigate further to make sure.